Outlook integration helm chart (#3599)
wireapp/wire-server: charts/outlook-addin/README.md

# How to install Outlook AddIn for Wire-Server

WIP: Some of these configurations are subject to change down the line. This documentation will be updated accordingly as they happen.

This document assumes you already have an instance of wire-server running. If you don't, follow this [documentation](https://github.com/wireapp/wire-server-deploy/blob/master/offline/docs.md)

> ## Set up OAuth with wire-server

Reviewer: ```suggestion
## Set up OAuth with Wire-Server
```

```
brig:
  # ...
  config:
    # ...
    optSettings:
      # ...
      setOAuthEnabled: true
```

Then you will need to generate a key using "OKP" (Octet Key Pair) and the "Ed25519" curve with OpenSSL that will be used as JWK (JSON Web Key) in the wire-server helm chart. This key will be used to sign and verify [OAuth](https://docs.wire.com/developer/reference/oauth.html#setting-up-public-and-private-keys) access tokens.

```
openssl genpkey -algorithm Ed25519 -out private_key.pem
```

You can find a `generate_jwk.py` in this chart which you can use to generate the JWK in JSON format that can be used in your wire-server helm chart. Use it in `brig` and `nginz` namespaces in `values/wire-server/secrets.yaml` like shown below.

```
brig:
  secrets:
    oauthJwkKeyPair: |
      {
        "kty": "OKP",
        "crv": "Ed25519",
        "x": "...",
        "d": "...",
        "kid": "..."
      }
```

```
# values.yaml or secrets.yaml
nginz:
  secrets:
    oAuth:
      publicKeys: |
        {
          "kty": "OKP",
          "crv": "Ed25519",
          "x": "...",
          "kid": "..."
        }
```

> Now redeploy wire-server chart:

Reviewer: ```suggestion
Now redeploy the Wire-Server chart:
```

```
d helm upgrade --install wire-server charts/wire-server --values values/wire-server/values.yaml --values/wire-server/secrets.yaml
```

## Outlook integration feature flag

By default, outlook addin as a feature is disabled for all teams. To change this make the following changes in your configuration in `galley` namespace:

```
galley:
  config:
    # ...
    settings:
      # ...
      featureFlags:
        # ...
        outlookCalIntegration:
          defaults:
            status: enabled
            lockStatus: unlocked
```

> Redeploy wire-server for these changes to take effect.

Reviewer: ```suggestion
Note that you must re-deploy wire-server for these changes to take effect.
```

NOTE: As of the time of writing `outlookCalIntegration` is not a typo! (at least not in this documentation)

> If you have an existing team in your wire-server that did not have this feature flag enabled prior to this. You will need to enable that feature flag through [Backoffice API](https://github.com/wireapp/wire-server/tree/05778a2b14ac5aaffca937d6e2cdd9b7b5f3106d/charts/backoffice).

Reviewer: ```suggestion
If you have an existing team in your Wire backend that did not have this feature flag enabled prior to this. You will need to enable that feature flag through [Backoffice API](https://github.com/wireapp/wire-server/tree/05778a2b14ac5aaffca937d6e2cdd9b7b5f3106d/charts/backoffice).
```
